libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.ts

```diff
@@ -376,66 +363,68 @@ export class SkyTextEditorComponent implements AfterViewInit, OnDestroy {
-      this.id,
-      (this.iframeRef as ElementRef).nativeElement,
-      this.initialStyleState,
-      this.placeholder
-    );
-
-    this.#editorService
-      .inputListener()
-      .pipe(takeUntil(this.#ngUnsubscribe))
-      .subscribe(() => {
-        // Angular doesn't run change detection for changes originating inside an iframe,
-        // so we have to call the onChange() event inside NgZone to force change propagation to consuming components.
-        this.#zone.run(() => {
-          this.#viewToModelUpdate();
+    if (this.iframeRef) {
+      this.#adapterService.initEditor(
+        this.id,
+        this.iframeRef,
+        this.initialStyleState,
+        this.placeholder
+      );
+
+      this.#editorService
+        .inputListener()
+        .pipe(takeUntil(this.#ngUnsubscribe))
+        .subscribe(() => {
+          // Angular doesn't run change detection for changes originating inside an iframe,
+          // so we have to call the onChange() event inside NgZone to force change propagation to consuming components.
+          this.#zone.run(() => {
+            this.#viewToModelUpdate();
+          });
         });
-      });
-
-    this.#editorService
-      .selectionChangeListener()
-      .pipe(takeUntil(this.#ngUnsubscribe))
-      .subscribe(() => {
-        this.#updateStyle();
-        this.editorFocusStream.next();
-      });
-
-    this.#editorService
-      .clickListener()
-      .pipe(takeUntil(this.#ngUnsubscribe))
-      .subscribe(() => {
-        this.editorFocusStream.next();
-      });
-
-    this.#editorService
-      .blurListener()
-      .pipe(takeUntil(this.#ngUnsubscribe))
-      .subscribe(() => {
-        // Angular doesn't run change detection for changes originating inside an iframe,
-        // so we have to run markForCheck() inside the NgZone to force change propagation to consuming components.
-        this.#zone.run(() => {
-          this.#_onTouched();
+
+      this.#editorService
+        .selectionChangeListener()
+        .pipe(takeUntil(this.#ngUnsubscribe))
+        .subscribe(() => {
+          this.#updateStyle();
+          this.editorFocusStream.next();
         });
-      });
 
-    this.#editorService
-      .commandChangeListener()
-      .pipe(takeUntil(this.#ngUnsubscribe))
-      .subscribe(() => {
-        this.#updateStyle();
-        this.#viewToModelUpdate();
-      });
+      this.#editorService
+        .clickListener()
+        .pipe(takeUntil(this.#ngUnsubscribe))
+        .subscribe(() => {
+          this.editorFocusStream.next();
+        });
 
-    this.#adapterService.setEditorInnerHtml(this.#_value);
+      this.#editorService
+        .blurListener()
+        .pipe(takeUntil(this.#ngUnsubscribe))
+        .subscribe(() => {
+          // Angular doesn't run change detection for changes originating inside an iframe,
+          // so we have to run markForCheck() inside the NgZone to force change propagation to consuming components.
```

Comment:
Should `markForCheck` in the comment be `#_onTouched`?

Reply:
Updated - this was an existing comment - but does seem that it needed updated. Done